base/iterators.jl

```diff
@@ -240,6 +240,32 @@ end
 iteratorsize{I1,I2}(::Type{Zip{I1,I2}}) = zip_iteratorsize(iteratorsize(I1),iteratorsize(I2))
 iteratoreltype{I1,I2}(::Type{Zip{I1,I2}}) = and_iteratoreltype(iteratoreltype(I1),iteratoreltype(I2))
 
+# unzip
+```jldoctest
+
+For iterable objects of the same length, returns iterable sets, where the ith set contains
+the ith component of each input iterable object.
+
+Note that the inverse of [`unzip`](@ref) is [`zip`](@ref).
+
+julia>  z = [[1,"a",:meow], [2,"b",:woof], [3,"c",:doh!]]
+3-element Array{Array{Any,1},1}:
+Any[1, "a", :meow]
+Any[2, "b", :woof]
+Any[3, "c", :doh!]
+
+julia> unzip(z)
+Array{Int64,1}:[1,2,3]
+Array{String,1}["a","b","c"]
+Array{Symbol,1}[:meow,:woof,:doh!]
+```
+unzip(iter::AbstractZipIterator) = _unzip(iter)
+# This avoids exporting the method that isn't for AbstractZipIterators
+_unzip(iter::Zip1) = (iter.a,)
+_unzip(iter::Zip2) = (iter.a, iter.b)
+_unzip(iter::Zip) = (iter.a, unzip(iter.z)...)
```

Comment:
The inner call to `unzip` here should actually be `_unzip`. (That was my typo initially, sorry about that.)